### PR TITLE
grpc-service: Add_split_and_scatter_region

### PR DESCRIPTION
**Pull request by hzh0425**
ref: https://github.com/pingcap/tidb/issues/29034

Add a method splitAndScatterRegions in grpc-service
let pd handle split and scatter regions

### CODE REVIEW

**Review by disksing — APPROVED** [review]

File: client/client.go [hunk 1]
```diff
@@ -107,6 +107,8 @@ type Client interface {
 	ScatterRegions(ctx context.Context, regionsID []uint64, opts ...RegionsOption) (*pdpb.ScatterRegionResponse, error)
 	// SplitRegions split regions by given split keys
 	SplitRegions(ctx context.Context, splitKeys [][]byte, opts ...RegionsOption) (*pdpb.SplitRegionsResponse, error)
+	// splitAndScatterRegions split regions by given split keys and scatter new regions
```
Comment (youjiali1995) [inline client/client.go:110]:
```suggestion
	// SplitAndScatterRegions split regions by given split keys and scatter new regions
```

File: client/client.go [hunk 2]
```diff
@@ -1695,6 +1697,32 @@ func (c *client) scatterRegionsWithOptions(ctx context.Context, regionsID []uint
 	return resp, nil
 }
 
+func (c *client) SplitAndScatterRegions(ctx context.Context, splitKeys [][]byte, opts ...RegionsOption) (*pdpb.SplitAndScatterRegionsResponse, error) {
+	start := time.Now()
+	defer func() { cmdDurationScatterRegions.Observe(time.Since(start).Seconds()) }()
```
Comment (youjiali1995) [inline client/client.go:1702]: Is it suitable to use `cmdDurationScatterRegions` for `SplitAndScatterRegions`? Maybe you need to add `cmdDurationSplitAndScatterRegions` for it.

File: go.mod [hunk 3]
```diff
@@ -4,7 +4,6 @@ go 1.16
 
 require (
 	github.com/BurntSushi/toml v0.3.1
-	github.com/alecthomas/template v0.0.0-20190718012654-fb15b899a751
```
Comment (youjiali1995) [inline go.mod:7]: Please run `go mod tidy` and commit the change.
```
go mod tidy
GO111MODULE=on go mod tidy
git diff --quiet go.mod go.sum
make: *** [Makefile:217: tidy] Error 1
```

File: server/grpc_service.go [hunk 4]
```diff
@@ -1555,6 +1555,55 @@ func (s *Server) SplitRegions(ctx context.Context, request *pdpb.SplitRegionsReq
 	}, nil
 }
 
+// SplitAndScatterRegions split regions by the given split keys, and scatter regions
+func (s *Server) SplitAndScatterRegions(ctx context.Context, request *pdpb.SplitAndScatterRegionsRequest) (*pdpb.SplitAndScatterRegionsResponse, error) {
```
Comment (youjiali1995) [inline server/grpc_service.go:1559]: Any test for this service and client?

**Review by youjiali1995 — COMMENTED** [review]
I can't update kvproto for TiDB because PD hasn't implement the interface yet.😭

**Pull request closed without merging**